Packages/KeyAppKit/Sources/BankTransfer/Striga/Repository/StrigaBankTransferUserDataRepository.swift

```diff
@@ -421,17 +421,44 @@ private extension UserWallet {
     }
 }
 
-extension StrigaBankTransferUserDataRepository {
-
+public extension StrigaBankTransferUserDataRepository {
     // MARK: - Withdrawal
 
-    public func withdrawalInfo() async throws -> WithdrawalInfo? {
-        await localProvider.getCachedWithdrawalInfo()// ??
-        /// GetAccountStatement here
-//        WithdrawalInfo(IBAN: "IBAN", BIC: "BIC", receiver: "Receiver")
+    func getWithdrawalInfo(userId: String) async throws -> StrigaWithdrawalInfo? {
+        if let cached = await localProvider.getCachedWithdrawalInfo() {
+            return cached
+        } else {
+            guard let accountId = await localProvider.getCachedUserData()?.wallet?.accounts.eur?.accountID else {
+                throw StrigaProviderError.missingAccountId
+            }
+
+            let transactions = try await remoteProvider.getAccountStatement(
+                userId: userId,
+                accountId: accountId,
+                startDate: Constants.startDate,
+                endDate: Date(),
+                page: 1
+            ).transactions
+            let completedTransaction = transactions.first(where: { $0.txType == Constants.sepaPayoutCompleted })
+            let initialTransaction = transactions
+                .first(where: { $0.id == completedTransaction?.id && $0.txType == Constants.sepaPayoutInitiated }) ??
+                transactions.first(where: { $0.txType == Constants.sepaPayinCompleted })
```

Comment:
seems like you can move $0.id == completedTransaction?.id && $0.txType == Constants.sepaPayoutInitiated to a previous `first(where:` block